client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -89,6 +89,7 @@ export class DomainDetailsForm extends PureComponent {
 
 		this.state = {
 			form: null,
+			hasAutoFilledContactDetails: !! this.props.contactDetails,
```

Comment:
Seems I was wrong. We always return something from the back-end, and prefill it with the user info we have on file. You can remove this.